### PR TITLE
migrate from getopts to clap in `split`

### PR DESCRIPTION
**Pull request by FelipeLema**
mind that those TODO I added are markers for ongoing development

### CODE REVIEW

File: src/uu/split/src/split.rs [hunk 1]
```diff
@@ -7,11 +7,15 @@
 
 // spell-checker:ignore (ToDO) PREFIXaa
 
+#[macro_use]
+extern crate clap;
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:11]: I don't think it is needed anymore 
See https://github.com/uutils/coreutils/pull/1673/files

File: src/uu/split/src/split.rs [hunk 2]
```diff
@@ -20,80 +24,122 @@ use std::path::Path;
...
-        Ok(m) => m,
-        Err(f) => crash!(1, "{}", f),
-    };
-
-    if matches.opt_present("h") {
-        let msg = format!(
-            "{0} {1}
-
-Usage:
+// TODO: pack OPTION_… into local module?
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:27]: Can you do that in alphabetic order please ? thanks!

File: src/uu/split/src/split.rs [hunk 3]
```diff
@@ -20,80 +24,122 @@ use std::path::Path;
...
+static ARG_INPUT: &str = "input";
+static ARG_PREFIX: &str = "prefix";
+
+fn get_usage() -> String {
+    format!("{0} [OPTION]... [INPUT [PREFIX]]", executable!())
+}
+fn get_long_usage() -> String {
+    String::from(
+        "Usage:
   {0} [OPTION]... [INPUT [PREFIX]]
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:47]: I think "{0} [OPTION]... [INPUT [PREFIX]]" is a duplicate as you have this in get_usage() too?

File: src/uu/split/src/split.rs [hunk 4]
```diff
@@ -20,80 +24,122 @@ use std::path::Path;
...
+    let usage = get_usage();
+    let long_usage = get_long_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about("Create output files containing consecutive or interleaved sections of input")
+        .usage(&usage[..])
+        .after_help(&long_usage[..])
+        .arg(
+            Arg::with_name(OPTION_SUFFIX_LENGTH)
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:65]: alphabetic order too please

File: src/uu/split/src/split.rs [hunk 5]
```diff
@@ -20,80 +24,122 @@ use std::path::Path;
...
+                .help("use suffixes of length N (default 2)"),
+        )
+        .arg(
+            Arg::with_name(OPTION_BYTES)
+                .short("b")
+                .conflicts_with(OPTION_LINES)
+                .conflicts_with(OPTION_LINE_BYTES)
+                .long(OPTION_BYTES)
+                .takes_value(true)
+                .default_value("2")
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:79]: does it has to be a string ?
it cannot be 2 directly ?

**Review by sylvestre — COMMENTED** [review]
Excellent, thanks!

File: src/uu/split/src/split.rs (hunk 1 above)
Comment (FelipeLema) [reply src/uu/split/src/split.rs:11]: that's cool, that's cool... will remove it

File: src/uu/split/src/split.rs (hunk 2 above)
Comment (FelipeLema) [reply src/uu/split/src/split.rs:27]: fo sho

File: src/uu/split/src/split.rs (hunk 3 above)
Comment (FelipeLema) [reply src/uu/split/src/split.rs:47]: right, totally updating it

File: src/uu/split/src/split.rs (hunk 4 above)
Comment (FelipeLema) [reply src/uu/split/src/split.rs:65]: fo sho

File: src/uu/split/src/split.rs (hunk 5 above)
Comment (FelipeLema) [reply src/uu/split/src/split.rs:79]: I scheemed for this rather quickly and thought that it couldn't be done... do you have an example where a typed arg is used (is it called "typed arg")?

Comment (sylvestre) [reply src/uu/split/src/split.rs:79]: you are correct, my bad, sorry!

File: src/uu/split/src/split.rs [hunk 6]
```diff
@@ -22,111 +19,112 @@ use std::fs::File;
...
-static OPTION_FILTER: &str = "filter";
-static OPTION_LINES: &str = "lines";
-static OPTION_VERBOSE: &str = "verbose";
-
-static ARG_INPUT: &str = "input";
-static ARG_PREFIX: &str = "prefix";
+const NAME: &str = "split";
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+mod options {
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:25]: Well, I like this approach but I prefer consistency with the rest of the binaries ... :/

I have been trying to make them closer to each others in term of styles to facilitate contributions but this isn't the case here.

Comment (FelipeLema) [reply src/uu/split/src/split.rs:25]: I changed this because I wanted to do `match` on a couple of strings... will try another way around it, then

Comment (sylvestre) [reply src/uu/split/src/split.rs:25]: hmm, you marked it as resolved but I still see the "mod options" ?
thanks

Comment (sylvestre) [reply src/uu/split/src/split.rs:25]: to be clear, I was talking about this kind of declaration:
https://github.com/uutils/coreutils/blob/master/src/uu/mktemp/src/mktemp.rs#L30-L35

but I think yours is much better.

Would you be interested in updating the rest of the coreutils to your mod options { } style?

Comment (FelipeLema) [reply src/uu/split/src/split.rs:25]: oh, I misunderstood you... I thought you meant the `const` bit

I'll change it back to `static OPTION_`

Also, I'll do a separate pull request for `mod options { }`

Comment (sylvestre) [reply src/uu/split/src/split.rs:25]: Excellent, thanks!

File: src/uu/split/src/split.rs [hunk 7]
```diff
@@ -22,19 +22,15 @@ use std::path::Path;
...
-        pub static LINES: &str = "lines";
-    }
-    pub static ADDITIONAL_SUFFIX: &str = "additional-suffix";
-    pub static FILTER: &str = "filter";
-    pub static NUMERIC_SUFFIXES: &str = "numeric-suffixes";
-    pub static SUFFIX_LENGTH: &str = "suffix-length";
-    pub static DEFAULT_SUFFIX_LENGTH: usize = 2;
-    pub static VERBOSE: &str = "verbose";
-}
+static OPTION_BYTES: &str = "bytes";
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:25]: sorry to be a pain but s/OPTION_/OPT_/g to be consistent with the other programs :)

Comment (FelipeLema) [reply src/uu/split/src/split.rs:25]: oh, sorry about that... will change it

**Pull request merged**